File: herepark-backend/app/schemas/parking.py
from datetime import datetime

from pydantic import BaseModel


class Position(BaseModel):
    lat: float
    lng: float


class ReservationInformation(BaseModel):
    reservation_id: int
    reservation_datetime: datetime


class ParkingSpaceInformation(BaseModel):
    space_id: int
    position: Position
    average_rating: str
    reservations: list[ReservationInformation]


class ParkingSpaceForm(BaseModel):
    latitude: float
    longitude: float
    reservation_limit: int


class ReservationForm(BaseModel):
    space_id: int
    reservation_datetime: datetime


class RatingForm(BaseModel):
    space_id: int
    rating: int

File: herepark-frontend/src/schemas/parking.ts
export interface Position {
    lat: number
    lng: number
}

export interface ReservationInformation {
    reservation_id: number
    reservation_datetime: Date
}

export interface ParkingSpaceInformation {
    space_id: number
    position: Position
    average_rating: string
    reservations: ReservationInformation[]
}

export interface ReservationForm {
    space_id: number
    reservation_datetime: Date
}

export interface RatingForm {
    space_id: number
    rating: number
}
